File: src/main/java/com/example/robotcontrolsystembackend/domain/model/Device.java
package com.example.robotcontrolsystembackend.domain.model;

import com.example.robotcontrolsystembackend.domain.enumtype.ConnectionType;
import com.example.robotcontrolsystembackend.domain.enumtype.DeviceStatus;
import com.example.robotcontrolsystembackend.domain.enumtype.DeviceType;
import com.example.robotcontrolsystembackend.domain.enumtype.RobotType;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

@Entity
@Table(name = "device")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Device {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "device_id")
    private Long deviceId;

    @Column(name = "hub_id", nullable = false)
    private Long hubId;

    @Column(name = "device_name", nullable = false, length = 150)
    private String deviceName;

    @Enumerated(EnumType.STRING)
    @Column(name = "device_type", nullable = false, columnDefinition = "device_type_enum")
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    private DeviceType deviceType;

    @Enumerated(EnumType.STRING)
    @Column(name = "robot_type", columnDefinition = "robot_type_enum")
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    private RobotType robotType;

    @Column(name = "model", length = 100)
    private String model;

    @Column(name = "serial_number", unique = true, length = 100)
    private String serialNumber;

    @Enumerated(EnumType.STRING)
    @Column(name = "connection_type", nullable = false, columnDefinition = "connection_type_enum")
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    private ConnectionType connectionType;

    @Enumerated(EnumType.STRING)
    @Column(name = "device_status", nullable = false, columnDefinition = "device_status_enum")
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    private DeviceStatus deviceStatus;
}

File: mo_src/lib/models/device.dart
class Device {
  final String id;
  final String hubId;
  final String name;
  final String deviceType;
  final String robotType;
  final String model;
  final String serialNumber;
  final String connectionType;
  final String status;

  Device({
    required this.id,
    required this.hubId,
    required this.name,
    required this.deviceType,
    required this.robotType,
    required this.model,
    required this.serialNumber,
    required this.connectionType,
    required this.status,
  });

  factory Device.fromJson(Map<String, dynamic> json) {
    return Device(
      id: (json['deviceId'] ?? json['device_id'] ?? '').toString(),
      hubId: (json['hubId'] ?? json['hub_id'] ?? '').toString(),
      name: (json['deviceName'] ?? json['device_name'] ?? 'No Name').toString(),
      deviceType: (json['deviceType'] ?? json['device_type'] ?? '').toString(),
      robotType: (json['robotType'] ?? json['robot_type'] ?? '').toString(),
      model: (json['model'] ?? '').toString(),
      serialNumber: (json['serialNumber'] ?? json['serial_number'] ?? '')
          .toString(),
      connectionType: (json['connectionType'] ?? json['connection_type'] ?? '')
          .toString(),
      status:
          (json['deviceStatus'] ?? json['device_status'] ?? 'Active').toString(),
    );
  }
}
